src/fsharp/NameResolution.fs

```diff
@@ -4380,26 +4380,24 @@ let rec ResolvePartialLongIdentInModuleOrNamespaceForItem (ncenv: NameResolver)
     }
 
 let rec PartialResolveLookupInModuleOrNamespaceAsModuleOrNamespaceThenLazy f plid (modref: ModuleOrNamespaceRef) =
-    let mty = modref.ModuleOrNamespaceType
     match plid with 
     | [] -> f modref
     | id :: rest -> 
-        match mty.ModulesAndNamespacesByDemangledName.TryFind id with
+        match modref.ModuleOrNamespaceType.ModulesAndNamespacesByDemangledName.TryFind id with
         | Some mty -> 
-            PartialResolveLookupInModuleOrNamespaceAsModuleOrNamespaceThenLazy f rest (modref.NestedTyconRef mty) 
+            modref.NestedTyconRef mty
+            |> PartialResolveLookupInModuleOrNamespaceAsModuleOrNamespaceThenLazy f rest
         | None -> Seq.empty
 
 let PartialResolveLongIndentAsModuleOrNamespaceThenLazy (nenv:NameResolutionEnv) plid f =
-    seq {
-        match plid with 
-        | id :: rest -> 
-            match Map.tryFind id nenv.eModulesAndNamespaces with
-            | Some modrefs -> 
-                for modref in modrefs do
-                    yield! PartialResolveLookupInModuleOrNamespaceAsModuleOrNamespaceThenLazy f rest modref
-            | None -> ()
-        | [] -> ()
-    }
+    match plid with 
```

Comment:
I'm not a fan of eliminating `seq { ... }` except if perf data says we have to.  I think the original code is better.